agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -677,6 +677,10 @@ bool monitor_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
         mon_rssi.conf_rx_rssi_notification_threshold_dbm =
             update->config().monitor_rx_rssi_notification_threshold_dbm;
         mon_rssi.conf_disable_initiative_arp = update->config().monitor_disable_initiative_arp;
+
+        // Mark the enable flag as "false" to force update in hostapd status.
+        // The status is polled every "MONITOR_DB_AP_POLLING_RATE_SEC" and update the value.
+        mon_db.set_hostapd_enabled(false);
```

Comment:
I would prefer to avoid this ugly w/a .
Even if it means removing  the operational_state checking in the son_slave and reading the configuration twice.
Remember this happens only in init and re-connection states.

Reply:
You are welcome to suggest a better suggestion.
I didn't want to add a message for that and this is a quite simple solution with comments explaining it.